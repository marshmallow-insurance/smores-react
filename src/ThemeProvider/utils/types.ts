import * as theme from '@mrshmllw/smores-foundations/build/web/variables.json'
import { Color } from 'theme'

type Flatten<T, Prefix extends string = ''> = {
  [K in keyof T & string]: T[K] extends Record<string, any>
    ? Flatten<T[K], `${Prefix}${K}.`>
    : `${Prefix}${K}`
}[keyof T & string]

export type NewColour = Flatten<(typeof theme)['color']>

const legacyColorMap = {
  marshmallowPink: 'interactive.primary.base', // old â†’ new
}

export type LegacyColorMap = keyof typeof legacyColorMap
