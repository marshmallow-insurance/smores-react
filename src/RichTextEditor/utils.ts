import { $createParagraphNode, LexicalNode, RootNode } from 'lexical'

export const appendNodes = (root: RootNode, nodes: LexicalNode[]) => {
  nodes
    .filter((node) => node.__type !== 'linebreak')
    .map((node) => {
      if (node.__type === 'text') {
        const paragraphNode = $createParagraphNode()
        paragraphNode.append(node)
        return paragraphNode
      }
      return node
    })
    .forEach((node) => root.append(node))
}
