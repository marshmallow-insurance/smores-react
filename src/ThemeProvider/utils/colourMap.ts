import * as designTokens from '@mrshmllw/smores-foundations/build/web/variables.json'
import { Color } from 'theme'

type Flatten<T, Prefix extends string = ''> = {
  [K in keyof T & string]: T[K] extends Record<string, any>
    ? Flatten<T[K], `${Prefix}${K}.`>
    : `${Prefix}${K}`
}[keyof T & string]

export type NewColor = Flatten<
  Pick<typeof designTokens, 'color' | 'extended1' | 'thirdParty'>
>

export type ColorTypes = Color | NewColor

// TODO: Test how well this works
// we currently access numerical colours like `theme.color.neutral[100]` or in the case of 000 colours, `theme.color.neutral['000']`
export const getThemeColor = (path: NewColor): string => {
  return path
    .split('.')
    .reduce((acc, key) => acc?.[key], designTokens as any) as string
}

// a function that returns a flattened dot notation string path to access the color value
export const formatDesignTokenColor = (): string => {
  const { color, extended1, thirdParty } = designTokens
  const colors = { color, extended1, thirdParty }

  if (colors === null) {
    return ''
  }

  const result: string[] = []
  const destructureNestedObject = (obj: any, currentPath: string) => {
    for (const key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        const value = obj[key]
        const newPath = currentPath ? `${currentPath}.${key}` : key
        if (typeof value === 'object' && value !== null) {
          destructureNestedObject(value, newPath)
        } else {
          result.push(newPath)
        }
      }
    }
  }
  destructureNestedObject(colors, '')
  return result.join(', ')
}

// old colour name â†’ new colour path // base token value
export const legacyColorMap: Record<Color, NewColor> = {
  fairyFloss: 'color.surface.brand.100', // palette.primary.040
  bubblegum: 'color.surface.brand.200', // palette.primary.060
  marshmallowPink: 'color.surface.brand.300', // palette.primary.100
  lollipop: 'color.surface.brand.400', // palatte.primary.120

  // Core Secondary
  chia: 'color.feedback.inactive.100', // palette.secondary.040
  sesame: 'color.text.subtle', // palette.secondary.060
  liquorice: 'color.text.base', // palette.secondary.100
  boba: 'color.text.contrast', // palette.secondary.120

  // Core Tertiary
  cream: `color.surface.base.000`, // palette.neutral.000
  coconut: 'color.surface.base.100', // palette.neutral.010
  mascarpone: 'color.surface.base.200', // palette.neutral.020
  custard: 'color.surface.base.300', // palette.neutral.040

  // Brand Secondary
  feijoa: 'color.illustration.accent1.100', // palette.brand1.060
  spearmint: 'color.illustration.accent1.200', // palette.brand1.100
  macaroon: 'color.illustration.accent2.100', // palette.brand2.060
  blueberry: 'color.illustration.accent2.200', // palette.brand2.100
  pistachio: 'color.illustration.accent3.200', // palette.brand3.100
  matcha: 'color.illustration.accent3.100', // palette.brand3.060
  caramel: 'color.illustration.accent4.200', // palette.brand4.100
  peanut: 'color.illustration.accent4.100', // palette.brand4.060
  marzipan: 'color.illustration.neutral.400', // palette.neutral.100
  oatmeal: 'color.illustration.neutral.300', // palette.neutral.060
  satsuma: 'extended1.20', // extended1.020

  // Traffic light
  watermelon: 'color.feedback.negative.100', // palette.negative.020
  strawberry: 'color.feedback.negative.200', // palette.negative.100
  apple: 'color.feedback.positive.200', // palette.positive.100
  mint: 'color.feedback.positive.100', // palette.positive.020
  lemon: 'color.feedback.notice.200', // palette.notice.100
  sherbert: 'color.feedback.notice.100', // palette.notice.020
  tangerine: 'extended1.100', // extended1.100

  // Third-party brand colours
  compareTheMarket: 'thirdParty.compareTheMarket',
  confused: 'thirdParty.confusedCom',
  onfido: 'thirdParty.onfido',
  x: 'thirdParty.twitter', // x rebrand not yet reflected in foundations
  premfina: 'thirdParty.premfina',
  checkout: 'thirdParty.checkout',
  meta: 'thirdParty.facebook', // Meta rebrand not yet reflected in foundations
  stripe: 'thirdParty.stripe',
  intercom: 'thirdParty.intercom',
  ravelin: 'thirdParty.ravelin',
  rac: 'thirdParty.rac',
  hometree: 'thirdParty.hometree',
} as const

/* usage example */
// prop change:
// change color prop type to be `color?: Color | NewColor`
// component usage:
// <Component color={'marshmallowPink'}/>
// OR
// <Component color={'interactive.primary.base'}/>
// inside component:
// const resolvedColor = color in legacyColorMap ? getThemeColor(legacyColorMap[color]) : getThemeColor(color)

// the above should resolve to #FF88C8
