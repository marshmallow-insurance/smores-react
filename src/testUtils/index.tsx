import React, { ReactElement } from 'react'
import { render, RenderOptions } from '@testing-library/react'
import { SmoresThemeProvider } from '../ThemeProvider/ThemeProvider'

const Providers = ({ children }: { children: React.ReactNode }) => {
  return <SmoresThemeProvider>{children}</SmoresThemeProvider>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
export * from '@testing-library/jest-dom/vitest'
export * from 'jest-styled-components'

export { customRender as render }
