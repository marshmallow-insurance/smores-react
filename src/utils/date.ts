const ukTimeFormat = new Intl.DateTimeFormat('en-GB', {
  timeZone: 'Europe/London',
})

export const convertToUkDate = (date: Date): Date => {
  const dateString = ukTimeFormat.format(date)
  const [day, month, year] = dateString.split('/')

  return new Date(Number(year), Number(month) - 1, Number(day))
}
