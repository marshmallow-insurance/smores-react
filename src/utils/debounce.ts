/**
 * Debounces a method.
 * @param fn - The function to be debounced.
 * @param delay - The debouncing delay.
 * @returns A debounced version of the function.
 */
export const debounce = <Args extends unknown[], R>(
  fn: (...args: Args) => R,
  delay: number,
): ((...args: Args) => void) => {
  let timeout: NodeJS.Timeout | undefined

  return (...args: Args): void => {
    if (timeout) {
      clearTimeout(timeout)
    }

    timeout = setTimeout(() => {
      fn(...args)
      timeout = undefined
    }, delay)
  }
}
