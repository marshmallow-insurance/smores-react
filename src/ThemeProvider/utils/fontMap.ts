import * as designTokens from '@mrshmllw/smores-foundations/build/web/variables.json'

import { Typo } from '../../Text'
import { Theme } from '../../ThemeProvider/ThemeProvider'
import { getFromObject } from '../../utils/getFromObject'

type Font = typeof designTokens.font
type FontKey = keyof Font
export type FontValueObject = Font[keyof Font][keyof Font[keyof Font]]

const fontValueKeys = [
  'fontFamily',
  'fontSize',
  'fontWeight',
  'lineHeight',
  'letterSpacing',
  'textCase',
]

type captionTypeKeys = [keyof Font['caption']]
type bodyTypeKeys = [keyof Font['body']]
type headingTypeKeys = [keyof Font['heading']]
type heroTypeKeys = [keyof Font['hero']]
type labelTypeKeys = [keyof Font['label']]
type linkTypeKeys = [keyof Font['link']]

export type FontTypeMap = {
  caption: captionTypeKeys
  label: labelTypeKeys
  body: bodyTypeKeys
  heading: headingTypeKeys
  hero: heroTypeKeys
  link: linkTypeKeys
}[keyof Font]

// type for the path to each font type in design tokens, e.g. 'font.hero.200'
export type FontPathMap = {
  [K in FontKey]: K extends 'caption'
    ? `font.${K}.${captionTypeKeys[number]}`
    : K extends 'body'
      ? `font.${K}.${bodyTypeKeys[number]}`
      : K extends 'heading'
        ? `font.${K}.${headingTypeKeys[number]}`
        : K extends 'hero'
          ? `font.${K}.${heroTypeKeys[number]}`
          : K extends 'label'
            ? `font.${K}.${labelTypeKeys[number]}`
            : never
}[FontKey]

export type TypoTypes = FontPathMap | Typo

// a function that returns a font value object based on either the new design token path or the legacy typo mapping
export const resolveToThemeFont = (
  typo: TypoTypes,
  theme: Theme,
): FontValueObject => {
  const fontPath = getTypoPath(typo) // always returns new design token path
  return getFromObject({
    obj: theme,
    path: fontPath,
  })
}

export const getTypoPath = (typo: TypoTypes) => {
  return typo in legacyFontStyleMapping
    ? legacyFontStyleMapping[typo as keyof typeof legacyFontStyleMapping]
    : typo
}

export const formatDesignTokenFont = (): string => {
  const { font } = designTokens
  if (font === null) {
    return ''
  }

  const result: string[] = []
  const destructureNestedObject = (obj: any, currentPath: string) => {
    for (const key in obj) {
      if (Object.prototype.hasOwnProperty.call(obj, key)) {
        const value = obj[key]

        const newPath = currentPath ? `${currentPath}.${key}` : key

        // TODO: instead of trimming, we should not add paths that end with font value keys
        const trimmedPath = fontValueKeys.some((key) =>
          newPath.endsWith(`.${key}`),
        )
          ? newPath.slice(0, newPath.lastIndexOf('.'))
          : newPath

        if (typeof value === 'object' && value !== null) {
          destructureNestedObject(value, trimmedPath)
        } else {
          result.push(trimmedPath)
        }
      }
    }
  }
  destructureNestedObject(font, 'font')
  return result.join(', ')
}

export const newFontStyles: Font = designTokens.font

export const legacyFontStyleMapping: Record<Typo, FontPathMap> = {
  'hero-alternate': 'font.hero.300',
  hero: 'font.hero.200',
  'heading-alternate': 'font.hero.100',
  'heading-large': 'font.heading.500',
  'heading-medium': 'font.heading.300',
  'heading-small': 'font.heading.200',
  'headline-regular': 'font.heading.100',
  'headline-small': 'font.heading.100', // query this
  'body-standfirst': 'font.body.300',
  'body-regular': 'font.body.200',
  'body-small': 'font.body.100',
  caption: 'font.caption.100',
  label: 'font.label.100',
}
