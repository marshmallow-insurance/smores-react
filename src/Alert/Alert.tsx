import React, { FC, ReactElement, useState } from 'react'
import styled, { css, useTheme } from 'styled-components'

import { MarginProps } from 'utils/space'
import { NewColor, resolveToThemeColor } from '../ThemeProvider/utils/colourMap'

import { Box } from '../Box'
import { IconStrict } from '../IconStrict'
import { Icon } from '../Icon'
import { Text } from '../Text'
import { Icons } from '../Icon/iconsList'
import { Button } from '../Button'
import { Link } from '../Link'

type AlertType = 'info' | 'fallback' | 'notice' | 'negative' | 'positive'

type CtaProps =
  | {
      ctaType: 'button'
      ctaAction: () => void
      ctaLabel: string
    }
  | {
      ctaType: 'link'
      ctaAction: string
      ctaLabel: string
    }
  | {
      ctaType?: undefined
      ctaAction?: undefined
      ctaLabel?: undefined
    }

type AlertProps = {
  type?: AlertType
  title?: string
  message: string | ReactElement<unknown>
  isDismissible?: boolean
} & CtaProps &
  MarginProps

type AlertTypeConfig = {
  backgroundColor: NewColor
  accentColor: NewColor
  icon: Icons
}

const typeConfig: Record<AlertType, AlertTypeConfig> = {
  info: {
    backgroundColor: 'color.feedback.informative.100',
    accentColor: 'color.feedback.informative.200',
    icon: 'circle-info-filled',
  },
  fallback: {
    backgroundColor: 'color.surface.base.100',
    accentColor: 'color.surface.base.400',
    icon: 'circle-info-filled',
  },
  notice: {
    backgroundColor: 'color.feedback.notice.100',
    accentColor: 'color.feedback.notice.200',
    icon: 'triangle-exclamation-solid',
  },
  negative: {
    backgroundColor: 'color.feedback.negative.100',
    accentColor: 'color.feedback.negative.200',
    icon: 'warning-circle',
  },
  positive: {
    backgroundColor: 'color.feedback.positive.100',
    accentColor: 'color.feedback.positive.200',
    icon: 'included',
  },
}

export const Alert: FC<AlertProps> = ({
  type = 'info',
  title,
  message,
  isDismissible = false,
  ctaType,
  ctaAction,
  ctaLabel,
  ...marginProps
}) => {
  const theme = useTheme()

  const backgroundColor = resolveToThemeColor(
    typeConfig[type].backgroundColor,
    theme,
  )
  const accentColor = resolveToThemeColor(typeConfig[type].accentColor, theme)
  const [alertClosed, setAlertClosed] = useState(false)

  return (
    <StyledAlert
      $backgroundColor={backgroundColor}
      $accentColor={accentColor}
      $alertClosed={alertClosed}
      {...marginProps}
    >
      <Icon
        render={typeConfig[type].icon}
        size={16}
        m={{ custom: 2 }}
        color="liquorice"
      />
      <Box
        flex
        direction="row"
        justifyContent="space-between"
        alignItems="flex-start"
        width="100%"
        gap="8px"
      >
        <Box>
          {title && (
            <Text mb={{ custom: '4px' }} typo="headline-regular">
              {title}
            </Text>
          )}
          {typeof message === 'string' ? (
            <Text mb={ctaType && '8px'}>{message}</Text>
          ) : (
            message
          )}
          {ctaType === 'button' && typeof ctaAction === 'function' && (
            <Button smallButton fallbackStyle onClick={() => ctaAction()}>
              {ctaLabel}
            </Button>
          )}
          {ctaType === 'link' && typeof ctaAction === 'string' && (
            <Link href={ctaAction}>{ctaLabel}</Link>
          )}
        </Box>
        {isDismissible && (
          <IconStrict
            render="cross"
            size={16}
            color="liquorice"
            onClick={() => setAlertClosed(true)}
          />
        )}
      </Box>
    </StyledAlert>
  )
}

interface IStyledAlert {
  $backgroundColor: string
  $accentColor: string
  $alertClosed: boolean
}

const StyledAlert = styled(Box)<IStyledAlert>(
  ({ $backgroundColor, $accentColor, $alertClosed }) => css`
    display: flex;
    width: 100%;
    max-width: 512px;
    min-width: min-content;
    padding: 12px 12px 12px 20px;
    align-items: flex-start;
    gap: 8px;
    border-radius: 8px 16px 16px 8px;
    background: ${$backgroundColor};
    position: relative;
    display: ${$alertClosed ? 'none' : 'flex'};

    &:before {
      width: 8px;
      height: 100%;
      position: absolute;
      background: ${$accentColor};
      content: '';
      top: 0;
      left: 0;
      border-radius: 16px 0 0 16px;
    }
  `,
)
