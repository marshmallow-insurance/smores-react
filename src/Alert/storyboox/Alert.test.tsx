import React from 'react'
import { expect, it } from 'vitest'
import { render } from '../../testUtils'
import { Alert } from '../Alert'
import { noop } from '../../utils/noop'

describe('Alert Component', () => {
  it('should render Info alert correctly', () => {
    const { container } = render(
      <Alert type="info" title="Info Alert" message="This is an info alert." />,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with button cta', () => {
    const { container } = render(
      <Alert
        type="fallback"
        ctaType="button"
        ctaAction={noop}
        ctaLabel="Click Me"
        title="Test Alert"
        message="This is a test alert."
      />,
    )
    expect(container).toMatchSnapshot()
  })

  it('should render correctly with link cta', () => {
    const { container } = render(
      <Alert
        type="notice"
        ctaType="link"
        ctaAction=""
        ctaLabel="Learn More"
        title="Test Alert"
        message="This is a test alert."
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
