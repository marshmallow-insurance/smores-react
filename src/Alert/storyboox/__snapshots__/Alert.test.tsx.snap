// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert Component > should render Info alert correctly 1`] = `
.c3 {
  display: flex;
  width: 100%;
}

.c4 {
  max-width: 512px;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  transform: rotate(0deg);
}

.c2 svg {
  color: #292924!important;
}

.c6 {
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c7 {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c0 {
  display: flex;
  width: 100%;
  max-width: 512px;
  min-width: min-content;
  padding: 12px 12px 12px 20px;
  align-items: flex-start;
  gap: 8px;
  border-radius: 8px 16px 16px 8px;
  background: #d0e3f2;
  position: relative;
  display: flex;
}

.c0:before {
  width: 8px;
  height: 100%;
  position: absolute;
  background: #4d92c9;
  content: '';
  top: 0;
  left: 0;
  border-radius: 16px 0 0 16px;
}

@media (min-width: 0px) {
  .c1 {
    margin: 2px;
  }
}

@media (min-width: 0px) {
  .c3 {
    flex-direction: row;
  }
}

@media (min-width: 0px) {
  .c3 {
    justify-content: space-between;
  }
}

@media (min-width: 0px) {
  .c3 {
    align-items: flex-start;
  }
}

@media (min-width: 0px) {
  .c3 {
    gap: 8px;
  }
}

@media (min-width: 0px) {
  .c5 {
    margin-bottom: 4px;
  }
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
      data-testid="circle-info-filled-container"
    >
      <svg
        fill="none"
        height="100%"
        viewBox="0 0 16 16"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16C12.4187 16 16 12.4187 16 8C16 3.58125 12.4187 0 8 0C3.58125 0 0 3.58125 0 8C0 12.4187 3.58125 16 8 16ZM7 5C7 4.44688 7.44688 4 8 4C8.55313 4 9 4.44688 9 5C9 5.55312 8.55313 6 8 6C7.44688 6 7 5.55312 7 5ZM6.75 7H8.25C8.66562 7 9 7.33437 9 7.75V10.5H9.25C9.66562 10.5 10 10.8344 10 11.25C10 11.6656 9.66562 12 9.25 12H6.75C6.33437 12 6 11.6656 6 11.25C6 10.8344 6.33437 10.5 6.75 10.5H7.5V8.5H6.75C6.33437 8.5 6 8.16562 6 7.75C6 7.33437 6.33437 7 6.75 7Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5 c6"
          cursor="inherit"
          title=""
        >
          Info Alert
        </p>
        <p
          class="c7"
          cursor="inherit"
          title=""
        >
          This is an info alert.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Component > should render correctly with button cta 1`] = `
.c3 {
  display: flex;
  width: 100%;
}

.c4 {
  max-width: 512px;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  transform: rotate(0deg);
}

.c2 svg {
  color: #292924!important;
}

.c6 {
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c8 {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c9 {
  position: relative;
  background-color: #ff88c8;
  box-shadow: none;
  color: #292924;
  padding: 0 20px;
  outline: none;
  border-radius: 10000px;
  font-weight: 500;
  cursor: pointer;
  line-height: 100%;
  font-size: 16px;
  opacity: 1;
  width: auto;
  background-color: #ffffff;
  padding: 0 10px;
  min-width: 54px;
  font-size: 14px;
}

.c9:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c9:hover {
  background-color: #fbf8f5;
}

.c9:active {
  background-color: #f8f2ea;
}

.c9 .childrenContainer {
  padding: 9px 0;
}

.c9 span {
  margin: 0 5px 0 0;
}

.c10 {
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 1;
}

.c11 {
  padding: 16px 0;
  flex-grow: 1;
}

.c0 {
  display: flex;
  width: 100%;
  max-width: 512px;
  min-width: min-content;
  padding: 12px 12px 12px 20px;
  align-items: flex-start;
  gap: 8px;
  border-radius: 8px 16px 16px 8px;
  background: #fbf8f5;
  position: relative;
  display: flex;
}

.c0:before {
  width: 8px;
  height: 100%;
  position: absolute;
  background: #dad2c4;
  content: '';
  top: 0;
  left: 0;
  border-radius: 16px 0 0 16px;
}

@media (min-width: 0px) {
  .c1 {
    margin: 2px;
  }
}

@media (min-width: 0px) {
  .c3 {
    flex-direction: row;
  }
}

@media (min-width: 0px) {
  .c3 {
    justify-content: space-between;
  }
}

@media (min-width: 0px) {
  .c3 {
    align-items: flex-start;
  }
}

@media (min-width: 0px) {
  .c3 {
    gap: 8px;
  }
}

@media (min-width: 0px) {
  .c5 {
    margin-bottom: 4px;
  }
}

@media (min-width: 0px) {
  .c7 {
    margin-bottom: 8px;
  }
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
      data-testid="circle-info-filled-container"
    >
      <svg
        fill="none"
        height="100%"
        viewBox="0 0 16 16"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16C12.4187 16 16 12.4187 16 8C16 3.58125 12.4187 0 8 0C3.58125 0 0 3.58125 0 8C0 12.4187 3.58125 16 8 16ZM7 5C7 4.44688 7.44688 4 8 4C8.55313 4 9 4.44688 9 5C9 5.55312 8.55313 6 8 6C7.44688 6 7 5.55312 7 5ZM6.75 7H8.25C8.66562 7 9 7.33437 9 7.75V10.5H9.25C9.66562 10.5 10 10.8344 10 11.25C10 11.6656 9.66562 12 9.25 12H6.75C6.33437 12 6 11.6656 6 11.25C6 10.8344 6.33437 10.5 6.75 10.5H7.5V8.5H6.75C6.33437 8.5 6 8.16562 6 7.75C6 7.33437 6.33437 7 6.75 7Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5 c6"
          cursor="inherit"
          title=""
        >
          Test Alert
        </p>
        <p
          class="c7 c8"
          cursor="inherit"
          title=""
        >
          This is a test alert.
        </p>
        <button
          class="c9"
        >
          <div
            class="c10"
          >
            <div
              class="c11 childrenContainer"
            >
              Click Me
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert Component > should render correctly with link cta 1`] = `
.c3 {
  display: flex;
  width: 100%;
}

.c4 {
  max-width: 512px;
}

.c2 {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-shrink: 0;
  width: 16px;
  height: 16px;
  transform: rotate(0deg);
}

.c2 svg {
  color: #292924!important;
}

.c10 {
  outline: 0;
  display: inline-flex;
  flex-direction: row;
  font-size: 16px;
  line-height: 20px;
  font-weight: 500;
  text-decoration: underline;
  color: #292924;
  background: none;
  cursor: pointer;
}

.c10:focus-visible {
  outline: 2px solid #292924;
  outline-offset: 2px;
}

.c10:hover {
  color: #636768;
}

.c10:hover path {
  fill: #636768;
}

.c10:active {
  color: #292924;
}

.c10:active path {
  fill: #636768;
}

.c6 {
  font-size: 16px;
  font-weight: 500;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c6 .c9 {
  color: #292924;
}

.c6 .c9 path {
  fill: #292924;
}

.c6 .c9:hover {
  color: #0e0e0c;
}

.c6 .c9:hover path {
  fill: #0e0e0c;
}

.c8 {
  font-size: 14px;
  font-weight: 400;
  line-height: 20px;
  text-align: left;
  color: #292924;
}

.c8 .c9 {
  color: #292924;
}

.c8 .c9 path {
  fill: #292924;
}

.c8 .c9:hover {
  color: #0e0e0c;
}

.c8 .c9:hover path {
  fill: #0e0e0c;
}

.c0 {
  display: flex;
  width: 100%;
  max-width: 512px;
  min-width: min-content;
  padding: 12px 12px 12px 20px;
  align-items: flex-start;
  gap: 8px;
  border-radius: 8px 16px 16px 8px;
  background: #fcf0d0;
  position: relative;
  display: flex;
}

.c0:before {
  width: 8px;
  height: 100%;
  position: absolute;
  background: #f2cb4a;
  content: '';
  top: 0;
  left: 0;
  border-radius: 16px 0 0 16px;
}

@media (min-width: 0px) {
  .c1 {
    margin: 2px;
  }
}

@media (min-width: 0px) {
  .c3 {
    flex-direction: row;
  }
}

@media (min-width: 0px) {
  .c3 {
    justify-content: space-between;
  }
}

@media (min-width: 0px) {
  .c3 {
    align-items: flex-start;
  }
}

@media (min-width: 0px) {
  .c3 {
    gap: 8px;
  }
}

@media (min-width: 0px) {
  .c5 {
    margin-bottom: 4px;
  }
}

@media (min-width: 0px) {
  .c7 {
    margin-bottom: 8px;
  }
}

<div>
  <div
    class="c0"
  >
    <span
      class="c1 c2"
      data-testid="triangle-exclamation-solid-container"
    >
      <svg
        fill="none"
        height="100%"
        viewBox="0 0 16 15"
        width="100%"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8.00065 0C8.46002 0 8.8819 0.253125 9.10065 0.65625L15.8506 13.1562C16.06 13.5437 16.0506 14.0125 15.8256 14.3906C15.6006 14.7688 15.1913 15 14.7506 15H1.25065C0.810021 15 0.400646 14.7688 0.175646 14.3906C-0.0493544 14.0125 -0.0587293 13.5437 0.150646 13.1562L6.90065 0.65625C7.1194 0.253125 7.54127 0 8.00065 0ZM8.00065 11C7.44752 11 7.00065 11.4469 7.00065 12C7.00065 12.5531 7.44752 13 8.00065 13C8.55377 13 9.00065 12.5531 9.00065 12C9.00065 11.4469 8.55377 11 8.00065 11ZM8.00065 5C7.4319 5 6.97877 5.48438 7.0194 6.05313L7.25065 9.30313C7.27877 9.69375 7.6069 10 7.99752 10C8.39127 10 8.71627 9.69688 8.7444 9.30313L8.97565 6.05313C9.01627 5.48438 8.56627 5 7.9944 5H8.00065Z"
          fill="currentColor"
        />
      </svg>
    </span>
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <p
          class="c5 c6"
          cursor="inherit"
          title=""
        >
          Test Alert
        </p>
        <p
          class="c7 c8"
          cursor="inherit"
          title=""
        >
          This is a test alert.
        </p>
        <a
          class="c9 c10"
          href=""
        >
          Learn More
        </a>
      </div>
    </div>
  </div>
</div>
`;
